zoom/__init__.py

```diff
@@ -33,34 +31,54 @@
 # possible.
 zoom_step = 2.0**0.25
 
+config = mw.addonManager.getConfig(__name__)
+
 
 def zoom_in(step=None):
     """Increase the text size."""
     if not step:
         step = zoom_step
-    mw.web.setZoomFactor(mw.web.zoomFactor() * step)
+    
+    change_zoom(mw.web.zoomFactor() * step)
 
 
 def zoom_out(step=None):
     """Decrease the text size."""
     if not step:
         step = zoom_step
-    mw.web.setZoomFactor(mw.web.zoomFactor() / step)
-
+    change_zoom(mw.web.zoomFactor() / step)
+    
+
+def set_zoom(state=None, *args):
+    """Set the zoom on state change"""
+    state = mw.state
+    
+    if state in ['deckBrowser', 'overview']:
+        mw.web.setZoomFactor( config[ 'overview_zoom' ] )
+    elif state == 'review':
+        mw.web.setZoomFactor( config[ 'review_zoom' ] )
+
+def change_zoom(new_zoom_level):
+    """When zoom is changed, save the values"""
+    state = mw.state
+    
+    if state in ['deckBrowser', 'overview']:
+        config[ 'overview_zoom' ] = new_zoom_level
+    elif state == 'review':
+        config[ 'review_zoom' ] = new_zoom_level
+    
+    mw.addonManager.writeConfig(__name__, config)
+    mw.web.setZoomFactor( new_zoom_level )
 
 def reset_zoom(state=None, *args):
     """Reset the text size."""
     if not state:
         state = mw.state
-    standard_zoom = deck_browser_standard_zoom
-    if 'overview' == state:
-        standard_zoom = overview_standard_zoom
-    if 'requestRequired' == state:
```

Comment:
Not sure about removal of this one - though I do not remember what it was supposed to do ;)